lib/mongoid/extensions/typed_array_class_factory.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Extensions
+
+    # A factory for manufacturing Mongoid::<Type>Array classes.
+    module TypedArrayClassFactory
+
+      class << self
+
+        # Create or retrieve the typed array class. If the class has not already
+        # been created, create a class called {Type}Array that inherits from
+        # Mongoid::TypedArray.
+        #
+        # @param [ Class ] type The type of the field.
+        #
+        # @return [ Class ] The typed array class.
+        #
+        # @api private
+        def create(type)
+          LOCK.synchronize do
+            const_string = "#{type}Array"
+            if Mongoid.const_defined?(const_string)
+              Mongoid.const_get(const_string)
+            else
+              array_class = typed_array_class(type)
+              array_class.const_set("Type", type)
+              Mongoid.const_set(const_string, array_class)
+            end
+          end
+        end
+
+        private
+
+        # @api private
+        LOCK = Mutex.new
```

Comment:
Declaring a `CONST` after `private` has no effect. Suggest to follow what seems to be Mongoid's code convention elsewhere to declare consts at/near top of class.

Reply:
No, this is a private constant and it doesn't need to be at the top.